### DISCUSSION TITLE
imported data and relationship based on existing columns

### DISCUSSION THREAD

**Discussion by KristofferBroberg** [Budibase/budibase#1562] [Help] [answered]
Hi!

First and foremost, I really feel Budibase is a great product which shows a lot of promise! It could very well save me and my partner from the hideous php crud I slapped together for her day job. (spent 60+ hours on something which in hindsight probably just should've been yet another spreadsheet)

Anyway, I understand that the capability to write to external sources is on your roadmap. In the meantime, is there a way to do  - basically-  an sql join on two internal tables? It isn't possible to create a relationship based on already existing imported data, right?

**Comment by mjashanks** [comment, marked as answer]
Hi Kristoffer.

External sources is already available - see the "External" tab in my screenshot below. We have some common datasources in there - in the coming months, we will add the ability to write your own connectors.

It is possible to add relationships between internal tables, although there is no way to import relationships - you would have to import your tables, and then recreate all the relationships, which is a painful process. It would be amazing to have the ability to understand and import relationships, but that sounds like a while away.

So using an external datasource is probably the way forward for you, given that you have existing relational data. 

![image](https://user-images.githubusercontent.com/3524181/119624301-e8cda200-be00-11eb-9ad3-845174ba6a4c.png)

**Reply by KristofferBroberg** [reply]
Hi! I forgot to mention that I went down the external sources path at first (reasonable when you have existing data) but I ran into trouble when trying to connect to my localhost db which is "outside" the docker container. I'll have a go at it again tonight.

Once again, wonderful product!
